src/services/ui/src/features/submission/app-k/index.tsx

```diff
@@ -41,14 +45,33 @@ export const AppKSubmissionForm = () => {
         user,
       },
       {
-        onSuccess: () => {
+        onSuccess: async () => {
           alert.setContent({
             header: "Package submitted",
             body: "The 1915(c) Appendix K Amendment Request has been submitted.",
           });
           alert.setBannerShow(true);
           alert.setBannerDisplayOn("/dashboard");
-          nav(originPath ? { path: originPath } : { path: "/dashboard" });
+          setIsDataPolling(true);
+          await new DataPoller({
+            interval: 1000,
+            pollAttempts: 20,
+            fetcher: () => getItem(`${draft.state}-${draft.parentWaiver}`),
+            checkStatus: (data) => {
+              return (
+                data._source.authority === Authority["1915c"] &&
+                data._source.actionType === "Amend"
+              );
+            },
+          }).startPollingData();
```

Comment:
we're using the poller a lot of places and values are pretty consistent for some parameters -- can we bundle any of this to isolate the point of control? especially while things are in flux moving forms around, as much consistency pre-built in is super helpful.

Reply:
(see below)